diff --git a/configure.py b/configure.py
index 1d0a1741ce..6d787e38c9 100755
--- a/configure.py
+++ b/configure.py
@@ -1242,8 +1242,8 @@ def configure_node(o):
   o['variables']['want_separate_host_toolset'] = int(cross_compiling)
 
   # Enable branch protection for arm64
-  if target_arch == 'arm64':
-    o['cflags']+=['-msign-return-address=all']
+#   if target_arch == 'arm64':
+#     o['cflags']+=['-msign-return-address=all']
 
   if options.node_snapshot_main is not None:
     if options.shared:
